dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/tests/Audacia.CodeAnalysis.Analyzers.Test/Rules/ControllerActionReturnTypedResultsAnalyzerTests.cs

```diff
@@ -0,0 +1,222 @@
+                {
+                    public class TestController : ControllerBase
+                    {
+                        [HttpGet]
+                        public Results<NotFound, Ok<string>> Get()
+                        {
+                            return 'hello';
+                        }
+                    }
+                }";
+
+            VerifyNoDiagnostic(testCode);
+        }
+
+        [TestMethod]
+        public void No_Diagnostics_For_Controller_With_IActionResults_ReturnType_Using_ProducesResponseType_Attribute()
+        {
+            const string testCode = @"
+                using System.Threading.Tasks;
+                using Microsoft.AspNetCore.Mvc;
+
+                namespace ConsoleApplication1
+                {
+                    public class TestController : ControllerBase
+                    {
+                        [HttpGet]
+                        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+                        public IActionResult Get()
+                        {
+                            return 'hello';
+                        }
+                    }
+                }";
+
+            VerifyNoDiagnostic(testCode);
+        }
+
+        [TestMethod]
+        public void No_Diagnostics_For_Controller_With_Async_Method_TypedResult_ReturnType_Not_Using_ProducesResponseType_Attribute()
+        {
+            const string testCode = @"
+                using System.Threading.Tasks;
+                using Microsoft.AspNetCore.Mvc;
+
+                namespace ConsoleApplication1
+                {
+                    public class TestController : ControllerBase
+                    {
+                        [HttpGet]
+                        public async Task<Results<NotFound, Ok<string>>> Get()
+                        {
+                            return 'hello';
+                        }
+                    }
+                }";
+
+            VerifyNoDiagnostic(testCode);
+        }
+
+        [TestMethod]
+        public void Diagnostics_For_Controller_With_TypedResults_ReturnType_And_ProducesResponseType_Attribute()
+        {
+            const string testCode = @"
+                using System.Threading.Tasks;
+                using Microsoft.AspNetCore.Mvc;
+
+                namespace ConsoleApplication1
+                {
+                    public class TestController : ControllerBase
+                    {
+                        [HttpGet]
+                        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+                        public Results<NotFound, Ok<string>> Get()
+                        {
+                            return 'hello';
+                        }
+                    }
+                }";
+
+            const string expectedMessage
+                = "Controller action name 'Get' [ProducesResponseType] attribute should not be applied when using TypedResults";
```

Comment:
This error message could be improved IMO, could we say something like:
> Controller action name 'Get' should not have a [ProducesResponseTypeAttribute] attribute if it returns TypedResults

?

Reply:
I was following what was in the AC, if you do think it fits better then I can change that but I do have to change the AC as well to match it,